### PR TITLE
[DOCS] EQL: Document `match` function

### PR DESCRIPTION
**Pull request by jrodewig**
Relates to #55182 and #55178

### CODE REVIEW

**Review by rw-access — APPROVED** [review]
LGTM

File: docs/reference/eql/functions.asciidoc [hunk 1]
```diff
@@ -416,6 +417,71 @@ field datatypes:
 *Returns:* integer or `null`
 ====
 
+[discrete]
+[[eql-fn-match]]
+=== `match`
+
+Returns `true` if a source string matches one or more provided regular
```
Comment (astefan) [inline docs/reference/eql/functions.asciidoc:424]: Do we know which regular expression is checked first? I mean, does it matter if it's matching the second or the fourth regular expression?

**Review by astefan — APPROVED** [review]
LGTM. Left one question, up to you if it needs clarification in docs or not. Thanks.

File: docs/reference/eql/functions.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/eql/functions.asciidoc:424]: I'm not sure it matters from a usage point of view.

The expressions are checked using `OR` logic so if _any_ of the regular expressions match, the function returns `true`.

The function also doesn't give any indicator of which expression matched. Only that at least one did.

`match("explorer.exe", "[a-z]*?.exe", "[1-9]")`
is equivalent to 
`match("explorer.exe", "[1-9]", "[a-z]*?.exe",)`

**Comment by jrodewig** [comment]
### Backport commits
master 7156f40 
7.x cd3663e

**Pull request merged**